javalib/src/main/scala/java/util/random/RandomGenerator.scala

```diff
@@ -0,0 +1,3 @@
+package java.util.random
+
+trait RandomGenerator
```

Comment:
This is introduced in JDK 17. I don't fully understand why it was included in the linkchain since it's not directly referenced, but adding this `trait` fixes it.

```
Error:  Found 1 missing definitions while linking
Error:  Not found Top(java.util.random.RandomGenerator)
Error:  	at /home/runner/work/scala-native/scala-native/unit-tests/shared/src/test/scala/javalib/util/SpittableRandomTest.scala:102
```

Reply:
@WojciechMazur thanks, but it still doesn't make sense to me.

1. Scala.js doesn't have this problem, even though the `RandomGenerator`interface is not implemented there.
```scala
//> using platform "js"
@main def main =
  val r = new java.util.SplittableRandom
  println(r.nextDouble())
```

2. The result of `javap -verbose` does not show any reference to `RandomGenerator`.

<details>

```
Classfile /workspace/scala-native/unit-tests/jvm/.2.13/target/scala-2.13/test-classes/org/scalanative/testsuite/javalib/util/SplittableRandomTest.class
  Last modified Oct 7, 2022; size 4111 bytes
  SHA-256 checksum bd382112573e7123c7fb00adf333dbad85521297b43f367944be217f69f36e7b
  Compiled from "SpittableRandomTest.scala"
public class org.scalanative.testsuite.javalib.util.SplittableRandomTest
  minor version: 0
  major version: 52
  flags: (0x0021) ACC_PUBLIC, ACC_SUPER
  this_class: #2                          // org/scalanative/testsuite/javalib/util/SplittableRandomTest
  super_class: #4                         // java/lang/Object
  interfaces: 0, fields: 0, methods: 6, attributes: 4
Constant pool:
    #1 = Utf8               org/scalanative/testsuite/javalib/util/SplittableRandomTest
    #2 = Class              #1            // org/scalanative/testsuite/javalib/util/SplittableRandomTest
    #3 = Utf8               java/lang/Object
    #4 = Class              #3            // java/lang/Object
    #5 = Utf8               SpittableRandomTest.scala
    #6 = Utf8               Lscala/reflect/ScalaSignature;
    #7 = Utf8               bytes
    #8 = Utf8               \u0006\u0005I2Aa\u0002\u0005\u0001\'!)!\u0004\u0001C\u00017!)a\u0004\u0001C\u0001?!)!\u0006\u0001C\u0001?!)A\u0006\u0001C\u0001?!)a\u0006\u0001C\u0001?!)\u0001\u0007\u0001C\u0001?\t!2\u000b\u001d7jiR\f\'\r\\3SC:$w.\u001c+fgRT!!\u0003\u0006\u0002\tU$\u0018\u000e\u001c\u0006\u0003\u00171\tqA[1wC2L\'M\u0003\u0002\u000e\u001d\u0005IA/Z:ugVLG/\u001a\u0006\u0003\u001fA\t1b]2bY\u0006t\u0017\r^5wK*\t\u0011#A\u0002pe\u001e\u001c\u0001a\u0005\u0002\u0001)A\u0011Q\u0003G\u0007\u0002-)\tq#A\u0003tG\u0006d\u0017-\u0003\u0002\u001a-\t1\u0011I\\=SK\u001a\fa\u0001P5oSRtD#\u0001\u000f\u0011\u0005u\u0001Q\"\u0001\u0005\u0002\u00119,\u0007\u0010\u001e\'p]\u001e$\u0012\u0001\t\t\u0003+\u0005J!A\t\f\u0003\tUs\u0017\u000e\u001e\u0015\u0003\u0005\u0011\u0002\"!\n\u0015\u000e\u0003\u0019R!a\n\t\u0002\u000b),h.\u001b;\n\u0005%2#\u0001\u0002+fgR\fqA\\3yi&sG\u000f\u000b\u0002\u0004I\u0005Qa.\u001a=u\t>,(\r\\3)\u0005\u0011!\u0013a\u00038fqR\u0014un\u001c7fC:D#!\u0002\u0013\u0002\u000bM\u0004H.\u001b;)\u0005\u0019!\u0003
    #9 = Utf8               nextLong
   #10 = Utf8               ()V
   #11 = Utf8               Lorg/junit/Test;
   #12 = Utf8               java/util/SplittableRandom
   #13 = Class              #12           // java/util/SplittableRandom
   #14 = Long               205620432625028l
   #16 = Utf8               <init>
   #17 = Utf8               (J)V
   #18 = NameAndType        #16:#17       // "<init>":(J)V
   #19 = Methodref          #13.#18       // java/util/SplittableRandom."<init>":(J)V
   #20 = Long               -546649510716590878l
   #22 = Utf8               ()J
   #23 = NameAndType        #9:#22        // nextLong:()J
   #24 = Methodref          #13.#23       // java/util/SplittableRandom.nextLong:()J
   #25 = Utf8               org/junit/Assert
   #26 = Class              #25           // org/junit/Assert
   #27 = Utf8               assertEquals
   #28 = Utf8               (JJ)V
   #29 = NameAndType        #27:#28       // assertEquals:(JJ)V
   #30 = Methodref          #26.#29       // org/junit/Assert.assertEquals:(JJ)V
   #31 = Long               5574037117696891406l
   #33 = Long               -2877648745898596966l
   #35 = Long               5734720902145206190l
   #37 = Long               1684781725002208217l
   #39 = Long               687902890032948154l
   #41 = Long               176280366443457561l
   #43 = Long               -2944062288620903198l
   #45 = Long               6872063775710978746l
   #47 = Long               -7374959378916621341l
   #49 = Long               3241340805431811560l
   #51 = Long               -2124831722811234979l
   #53 = Long               7339249063279462363l
   #55 = Long               1969867631102365324l
   #57 = Long               81632902222022867l
   #59 = Long               3451014011249622471l
   #61 = Long               -1727223780574897556l
   #63 = Long               -5128686556801302975l
   #65 = Long               -6412221907719417908l
   #67 = Long               -110482401893286265l
   #69 = Utf8               sr1
   #70 = Utf8               Ljava/util/SplittableRandom;
   #71 = Utf8               sr2
   #72 = Utf8               this
   #73 = Utf8               Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
   #74 = Utf8               nextInt
   #75 = Long               -84638l
   #77 = Long               962946964l
   #79 = Utf8               ()I
   #80 = NameAndType        #74:#79       // nextInt:()I
   #81 = Methodref          #13.#80       // java/util/SplittableRandom.nextInt:()I
   #82 = Long               1723227640l
   #84 = Long               -621790539l
   #86 = Long               -1848500421l
   #88 = Long               -614898617l
   #90 = Long               -628601850l
   #92 = Long               -463597391l
   #94 = Long               1874082924l
   #96 = Long               -1206032701l
   #98 = Long               1549874426l
  #100 = Long               -495782737l
  #102 = Long               -1487672352l
  #104 = Long               -538628223l
  #106 = Long               1117712970l
  #108 = Long               2081437683l
  #110 = Long               2134440938l
  #112 = Long               -2102672277l
  #114 = Long               832521577l
  #116 = Long               518494223l
  #118 = Long               -42114979l
  #120 = Utf8               nextDouble
  #121 = Long               -45l
  #123 = Double             0.8229662358649753d
  #125 = Utf8               ()D
  #126 = NameAndType        #120:#125     // nextDouble:()D
  #127 = Methodref          #13.#126      // java/util/SplittableRandom.nextDouble:()D
  #128 = Utf8               (DDD)V
  #129 = NameAndType        #27:#128      // assertEquals:(DDD)V
  #130 = Methodref          #26.#129      // org/junit/Assert.assertEquals:(DDD)V
  #131 = Double             0.43324117570991283d
  #133 = Double             0.2639712712295723d
  #135 = Double             0.5576376282289696d
  #137 = Double             0.5505810186639037d
  #139 = Double             0.3944509738261206d
  #141 = Double             0.3108138671457821d
  #143 = Double             0.585951421265481d
  #145 = Double             0.2009547438834305d
  #147 = Double             0.8317691736686829d
  #149 = Long               45l
  #151 = Double             0.9684135896502549d
  #153 = Double             0.9819686323309464d
  #155 = Double             0.5311927268453047d
  #157 = Double             0.8521356026917833d
  #159 = Double             0.01880601374789126d
  #161 = Double             0.37792881248018584d
  #163 = Double             0.7179744490511354d
  #165 = Double             0.3448879713662756d
  #167 = Double             0.023020123407108684d
  #169 = Double             0.6454709437764473d
  #171 = Utf8               nextBoolean
  #172 = Long               4782934l
  #174 = Utf8               ()Z
  #175 = NameAndType        #171:#174     // nextBoolean:()Z
  #176 = Methodref          #13.#175      // java/util/SplittableRandom.nextBoolean:()Z
  #177 = Utf8               assertFalse
  #178 = Utf8               (Z)V
  #179 = NameAndType        #177:#178     // assertFalse:(Z)V
  #180 = Methodref          #26.#179      // org/junit/Assert.assertFalse:(Z)V
  #181 = Utf8               assertTrue
  #182 = NameAndType        #181:#178     // assertTrue:(Z)V
  #183 = Methodref          #26.#182      // org/junit/Assert.assertTrue:(Z)V
  #184 = Long               -4782934l
  #186 = Utf8               split
  #187 = Utf8               ()Ljava/util/SplittableRandom;
  #188 = NameAndType        #186:#187     // split:()Ljava/util/SplittableRandom;
  #189 = Methodref          #13.#188      // java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
  #190 = Long               -2051870635339219700l
  #192 = Long               -4512002368431042276l
  #194 = Long               7607532382842316154l
  #196 = Long               -1011899051174066375l
  #198 = Long               -1531465968943756660l
  #200 = Long               948449286892387518l
  #202 = Long               2486448889230464769l
  #204 = Long               4550542803092639410l
  #206 = Long               8668601242423591169l
  #208 = Long               -986244092642826172l
  #210 = Long               274792684182118046l
  #212 = Long               683259797650761389l
  #214 = Long               1682793527903105269l
  #216 = Long               2140483520539013019l
  #218 = Long               -7468768144124082123l
  #220 = Long               6163667569279435512l
  #222 = Utf8               sr3
  #223 = Utf8               sr4
  #224 = Utf8               sr5
  #225 = Utf8               sr6
  #226 = Utf8               sr7
  #227 = Utf8               sr8
  #228 = NameAndType        #16:#10       // "<init>":()V
  #229 = Methodref          #4.#228       // java/lang/Object."<init>":()V
  #230 = Utf8               Code
  #231 = Utf8               LineNumberTable
  #232 = Utf8               LocalVariableTable
  #233 = Utf8               RuntimeVisibleAnnotations
  #234 = Utf8               SourceFile
  #235 = Utf8               ScalaInlineInfo
  #236 = Utf8               ScalaSig
{
  public void nextLong();
    descriptor: ()V
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=4, locals=3, args_size=1
         0: new           #13                 // class java/util/SplittableRandom
         3: dup
         4: ldc2_w        #14                 // long 205620432625028l
         7: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        10: astore_1
        11: ldc2_w        #20                 // long -546649510716590878l
        14: aload_1
        15: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        18: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        21: ldc2_w        #31                 // long 5574037117696891406l
        24: aload_1
        25: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        28: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        31: ldc2_w        #33                 // long -2877648745898596966l
        34: aload_1
        35: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        38: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        41: ldc2_w        #35                 // long 5734720902145206190l
        44: aload_1
        45: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        48: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        51: ldc2_w        #37                 // long 1684781725002208217l
        54: aload_1
        55: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        58: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        61: ldc2_w        #39                 // long 687902890032948154l
        64: aload_1
        65: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        68: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        71: ldc2_w        #41                 // long 176280366443457561l
        74: aload_1
        75: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        78: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        81: ldc2_w        #43                 // long -2944062288620903198l
        84: aload_1
        85: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        88: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        91: ldc2_w        #45                 // long 6872063775710978746l
        94: aload_1
        95: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        98: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       101: ldc2_w        #47                 // long -7374959378916621341l
       104: aload_1
       105: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       108: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       111: new           #13                 // class java/util/SplittableRandom
       114: dup
       115: ldc2_w        #47                 // long -7374959378916621341l
       118: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
       121: astore_2
       122: ldc2_w        #49                 // long 3241340805431811560l
       125: aload_2
       126: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       129: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       132: ldc2_w        #51                 // long -2124831722811234979l
       135: aload_2
       136: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       139: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       142: ldc2_w        #53                 // long 7339249063279462363l
       145: aload_2
       146: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       149: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       152: ldc2_w        #55                 // long 1969867631102365324l
       155: aload_2
       156: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       159: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       162: ldc2_w        #57                 // long 81632902222022867l
       165: aload_2
       166: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       169: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       172: ldc2_w        #59                 // long 3451014011249622471l
       175: aload_2
       176: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       179: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       182: ldc2_w        #61                 // long -1727223780574897556l
       185: aload_2
       186: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       189: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       192: ldc2_w        #63                 // long -5128686556801302975l
       195: aload_2
       196: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       199: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       202: ldc2_w        #65                 // long -6412221907719417908l
       205: aload_2
       206: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       209: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       212: ldc2_w        #67                 // long -110482401893286265l
       215: aload_2
       216: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       219: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       222: return
      LineNumberTable:
        line 25: 0
        line 26: 11
        line 27: 21
        line 28: 31
        line 29: 41
        line 30: 51
        line 31: 61
        line 32: 71
        line 33: 81
        line 34: 91
        line 35: 101
        line 37: 111
        line 38: 122
        line 39: 132
        line 40: 142
        line 41: 152
        line 42: 162
        line 43: 172
        line 44: 182
        line 45: 192
        line 46: 202
        line 47: 212
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           11     211     1   sr1   Ljava/util/SplittableRandom;
          122     100     2   sr2   Ljava/util/SplittableRandom;
            0     223     0  this   Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
    RuntimeVisibleAnnotations:
      0: #11()
        org.junit.Test

  public void nextInt();
    descriptor: ()V
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=4, locals=3, args_size=1
         0: new           #13                 // class java/util/SplittableRandom
         3: dup
         4: ldc2_w        #75                 // long -84638l
         7: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        10: astore_1
        11: ldc2_w        #77                 // long 962946964l
        14: aload_1
        15: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        18: i2l
        19: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        22: ldc2_w        #82                 // long 1723227640l
        25: aload_1
        26: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        29: i2l
        30: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        33: ldc2_w        #84                 // long -621790539l
        36: aload_1
        37: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        40: i2l
        41: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        44: ldc2_w        #86                 // long -1848500421l
        47: aload_1
        48: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        51: i2l
        52: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        55: ldc2_w        #88                 // long -614898617l
        58: aload_1
        59: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        62: i2l
        63: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        66: ldc2_w        #90                 // long -628601850l
        69: aload_1
        70: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        73: i2l
        74: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        77: ldc2_w        #92                 // long -463597391l
        80: aload_1
        81: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        84: i2l
        85: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        88: ldc2_w        #94                 // long 1874082924l
        91: aload_1
        92: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
        95: i2l
        96: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        99: ldc2_w        #96                 // long -1206032701l
       102: aload_1
       103: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       106: i2l
       107: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       110: ldc2_w        #98                 // long 1549874426l
       113: aload_1
       114: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       117: i2l
       118: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       121: new           #13                 // class java/util/SplittableRandom
       124: dup
       125: ldc2_w        #98                 // long 1549874426l
       128: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
       131: astore_2
       132: ldc2_w        #100                // long -495782737l
       135: aload_2
       136: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       139: i2l
       140: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       143: ldc2_w        #102                // long -1487672352l
       146: aload_2
       147: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       150: i2l
       151: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       154: ldc2_w        #104                // long -538628223l
       157: aload_2
       158: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       161: i2l
       162: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       165: ldc2_w        #106                // long 1117712970l
       168: aload_2
       169: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       172: i2l
       173: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       176: ldc2_w        #108                // long 2081437683l
       179: aload_2
       180: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       183: i2l
       184: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       187: ldc2_w        #110                // long 2134440938l
       190: aload_2
       191: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       194: i2l
       195: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       198: ldc2_w        #112                // long -2102672277l
       201: aload_2
       202: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       205: i2l
       206: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       209: ldc2_w        #114                // long 832521577l
       212: aload_2
       213: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       216: i2l
       217: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       220: ldc2_w        #116                // long 518494223l
       223: aload_2
       224: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       227: i2l
       228: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       231: ldc2_w        #118                // long -42114979l
       234: aload_2
       235: invokevirtual #81                 // Method java/util/SplittableRandom.nextInt:()I
       238: i2l
       239: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       242: return
      LineNumberTable:
        line 51: 0
        line 52: 11
        line 53: 22
        line 54: 33
        line 55: 44
        line 56: 55
        line 57: 66
        line 58: 77
        line 59: 88
        line 60: 99
        line 61: 110
        line 63: 121
        line 64: 132
        line 65: 143
        line 66: 154
        line 67: 165
        line 68: 176
        line 69: 187
        line 70: 198
        line 71: 209
        line 72: 220
        line 73: 231
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           11     231     1   sr1   Ljava/util/SplittableRandom;
          132     110     2   sr2   Ljava/util/SplittableRandom;
            0     243     0  this   Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
    RuntimeVisibleAnnotations:
      0: #11()
        org.junit.Test

  public void nextDouble();
    descriptor: ()V
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=6, locals=3, args_size=1
         0: new           #13                 // class java/util/SplittableRandom
         3: dup
         4: ldc2_w        #121                // long -45l
         7: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        10: astore_1
        11: ldc2_w        #123                // double 0.8229662358649753d
        14: aload_1
        15: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        18: dconst_0
        19: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        22: ldc2_w        #131                // double 0.43324117570991283d
        25: aload_1
        26: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        29: dconst_0
        30: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        33: ldc2_w        #133                // double 0.2639712712295723d
        36: aload_1
        37: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        40: dconst_0
        41: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        44: ldc2_w        #135                // double 0.5576376282289696d
        47: aload_1
        48: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        51: dconst_0
        52: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        55: ldc2_w        #137                // double 0.5505810186639037d
        58: aload_1
        59: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        62: dconst_0
        63: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        66: ldc2_w        #139                // double 0.3944509738261206d
        69: aload_1
        70: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        73: dconst_0
        74: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        77: ldc2_w        #141                // double 0.3108138671457821d
        80: aload_1
        81: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        84: dconst_0
        85: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        88: ldc2_w        #143                // double 0.585951421265481d
        91: aload_1
        92: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
        95: dconst_0
        96: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
        99: ldc2_w        #145                // double 0.2009547438834305d
       102: aload_1
       103: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       106: dconst_0
       107: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       110: ldc2_w        #147                // double 0.8317691736686829d
       113: aload_1
       114: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       117: dconst_0
       118: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       121: new           #13                 // class java/util/SplittableRandom
       124: dup
       125: ldc2_w        #149                // long 45l
       128: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
       131: astore_2
       132: ldc2_w        #151                // double 0.9684135896502549d
       135: aload_2
       136: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       139: dconst_0
       140: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       143: ldc2_w        #153                // double 0.9819686323309464d
       146: aload_2
       147: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       150: dconst_0
       151: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       154: ldc2_w        #155                // double 0.5311927268453047d
       157: aload_2
       158: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       161: dconst_0
       162: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       165: ldc2_w        #157                // double 0.8521356026917833d
       168: aload_2
       169: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       172: dconst_0
       173: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       176: ldc2_w        #159                // double 0.01880601374789126d
       179: aload_2
       180: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       183: dconst_0
       184: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       187: ldc2_w        #161                // double 0.37792881248018584d
       190: aload_2
       191: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       194: dconst_0
       195: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       198: ldc2_w        #163                // double 0.7179744490511354d
       201: aload_2
       202: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       205: dconst_0
       206: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       209: ldc2_w        #165                // double 0.3448879713662756d
       212: aload_2
       213: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       216: dconst_0
       217: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       220: ldc2_w        #167                // double 0.023020123407108684d
       223: aload_2
       224: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       227: dconst_0
       228: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       231: ldc2_w        #169                // double 0.6454709437764473d
       234: aload_2
       235: invokevirtual #127                // Method java/util/SplittableRandom.nextDouble:()D
       238: dconst_0
       239: invokestatic  #130                // Method org/junit/Assert.assertEquals:(DDD)V
       242: return
      LineNumberTable:
        line 77: 0
        line 78: 11
        line 79: 22
        line 80: 33
        line 81: 44
        line 82: 55
        line 83: 66
        line 84: 77
        line 85: 88
        line 86: 99
        line 87: 110
        line 89: 121
        line 90: 132
        line 91: 143
        line 92: 154
        line 93: 165
        line 94: 176
        line 95: 187
        line 96: 198
        line 97: 209
        line 98: 220
        line 99: 231
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           11     231     1   sr1   Ljava/util/SplittableRandom;
          132     110     2   sr2   Ljava/util/SplittableRandom;
            0     243     0  this   Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
    RuntimeVisibleAnnotations:
      0: #11()
        org.junit.Test

  public void nextBoolean();
    descriptor: ()V
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=4, locals=3, args_size=1
         0: new           #13                 // class java/util/SplittableRandom
         3: dup
         4: ldc2_w        #172                // long 4782934l
         7: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        10: astore_1
        11: aload_1
        12: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        15: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
        18: aload_1
        19: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        22: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
        25: aload_1
        26: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        29: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
        32: aload_1
        33: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        36: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
        39: aload_1
        40: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        43: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
        46: aload_1
        47: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        50: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
        53: aload_1
        54: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        57: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
        60: aload_1
        61: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        64: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
        67: aload_1
        68: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        71: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
        74: aload_1
        75: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        78: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
        81: new           #13                 // class java/util/SplittableRandom
        84: dup
        85: ldc2_w        #184                // long -4782934l
        88: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        91: astore_2
        92: aload_2
        93: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
        96: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
        99: aload_2
       100: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       103: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
       106: aload_2
       107: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       110: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
       113: aload_2
       114: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       117: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
       120: aload_2
       121: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       124: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
       127: aload_2
       128: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       131: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
       134: aload_2
       135: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       138: invokestatic  #180                // Method org/junit/Assert.assertFalse:(Z)V
       141: aload_2
       142: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       145: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
       148: aload_2
       149: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       152: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
       155: aload_2
       156: invokevirtual #176                // Method java/util/SplittableRandom.nextBoolean:()Z
       159: invokestatic  #183                // Method org/junit/Assert.assertTrue:(Z)V
       162: return
      LineNumberTable:
        line 103: 0
        line 104: 11
        line 105: 18
        line 106: 25
        line 107: 32
        line 108: 39
        line 109: 46
        line 110: 53
        line 111: 60
        line 112: 67
        line 113: 74
        line 115: 81
        line 116: 92
        line 117: 99
        line 118: 106
        line 119: 113
        line 120: 120
        line 121: 127
        line 122: 134
        line 123: 141
        line 124: 148
        line 125: 155
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           11     151     1   sr1   Ljava/util/SplittableRandom;
           92      70     2   sr2   Ljava/util/SplittableRandom;
            0     163     0  this   Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
    RuntimeVisibleAnnotations:
      0: #11()
        org.junit.Test

  public void split();
    descriptor: ()V
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=4, locals=9, args_size=1
         0: new           #13                 // class java/util/SplittableRandom
         3: dup
         4: ldc2_w        #14                 // long 205620432625028l
         7: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        10: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
        13: astore_1
        14: ldc2_w        #190                // long -2051870635339219700l
        17: aload_1
        18: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        21: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        24: ldc2_w        #192                // long -4512002368431042276l
        27: aload_1
        28: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        31: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        34: new           #13                 // class java/util/SplittableRandom
        37: dup
        38: ldc2_w        #192                // long -4512002368431042276l
        41: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        44: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
        47: astore_2
        48: ldc2_w        #194                // long 7607532382842316154l
        51: aload_2
        52: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        55: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        58: ldc2_w        #196                // long -1011899051174066375l
        61: aload_2
        62: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        65: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        68: new           #13                 // class java/util/SplittableRandom
        71: dup
        72: ldc2_w        #194                // long 7607532382842316154l
        75: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
        78: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
        81: astore_3
        82: ldc2_w        #198                // long -1531465968943756660l
        85: aload_3
        86: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        89: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
        92: ldc2_w        #200                // long 948449286892387518l
        95: aload_3
        96: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
        99: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       102: new           #13                 // class java/util/SplittableRandom
       105: dup
       106: ldc2_w        #200                // long 948449286892387518l
       109: invokespecial #19                 // Method java/util/SplittableRandom."<init>":(J)V
       112: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
       115: astore        4
       117: ldc2_w        #202                // long 2486448889230464769l
       120: aload         4
       122: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       125: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       128: ldc2_w        #204                // long 4550542803092639410l
       131: aload         4
       133: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       136: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       139: aload         4
       141: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
       144: astore        5
       146: ldc2_w        #206                // long 8668601242423591169l
       149: aload         5
       151: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       154: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       157: ldc2_w        #208                // long -986244092642826172l
       160: aload         5
       162: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       165: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       168: aload         4
       170: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
       173: astore        6
       175: ldc2_w        #210                // long 274792684182118046l
       178: aload         6
       180: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       183: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       186: ldc2_w        #212                // long 683259797650761389l
       189: aload         6
       191: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       194: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       197: aload         6
       199: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
       202: astore        7
       204: ldc2_w        #214                // long 1682793527903105269l
       207: aload         7
       209: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       212: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       215: ldc2_w        #216                // long 2140483520539013019l
       218: aload         7
       220: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       223: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       226: aload         6
       228: invokevirtual #189                // Method java/util/SplittableRandom.split:()Ljava/util/SplittableRandom;
       231: astore        8
       233: ldc2_w        #218                // long -7468768144124082123l
       236: aload         8
       238: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       241: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       244: ldc2_w        #220                // long 6163667569279435512l
       247: aload         8
       249: invokevirtual #24                 // Method java/util/SplittableRandom.nextLong:()J
       252: invokestatic  #30                 // Method org/junit/Assert.assertEquals:(JJ)V
       255: return
      LineNumberTable:
        line 129: 0
        line 130: 14
        line 131: 24
        line 133: 34
        line 134: 48
        line 135: 58
        line 137: 68
        line 138: 82
        line 139: 92
        line 141: 102
        line 142: 117
        line 143: 128
        line 145: 139
        line 146: 146
        line 147: 157
        line 149: 168
        line 150: 175
        line 151: 186
        line 153: 197
        line 154: 204
        line 155: 215
        line 157: 226
        line 158: 233
        line 159: 244
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           14     241     1   sr1   Ljava/util/SplittableRandom;
           48     207     2   sr2   Ljava/util/SplittableRandom;
           82     173     3   sr3   Ljava/util/SplittableRandom;
          117     138     4   sr4   Ljava/util/SplittableRandom;
          146     109     5   sr5   Ljava/util/SplittableRandom;
          175      80     6   sr6   Ljava/util/SplittableRandom;
          204      51     7   sr7   Ljava/util/SplittableRandom;
          233      22     8   sr8   Ljava/util/SplittableRandom;
            0     256     0  this   Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
    RuntimeVisibleAnnotations:
      0: #11()
        org.junit.Test

  public org.scalanative.testsuite.javalib.util.SplittableRandomTest();
    descriptor: ()V
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=1, locals=1, args_size=1
         0: aload_0
         1: invokespecial #229                // Method java/lang/Object."<init>":()V
         4: return
      LineNumberTable:
        line 22: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0       5     0  this   Lorg/scalanative/testsuite/javalib/util/SplittableRandomTest;
}
SourceFile: "SpittableRandomTest.scala"
RuntimeVisibleAnnotations:
  0: #6(#7=s#8)
    scala.reflect.ScalaSignature(
      bytes="\u0006\u0005I2Aa\u0002\u0005\u0001\'!)!\u0004\u0001C\u00017!)a\u0004\u0001C\u0001?!)!\u0006\u0001C\u0001?!)A\u0006\u0001C\u0001?!)a\u0006\u0001C\u0001?!)\u0001\u0007\u0001C\u0001?\t!2\u000b\u001d7jiR\f\'\r\\3SC:$w.\u001c+fgRT!!\u0003\u0006\u0002\tU$\u0018\u000e\u001c\u0006\u0003\u00171\tqA[1wC2L\'M\u0003\u0002\u000e\u001d\u0005IA/Z:ugVLG/\u001a\u0006\u0003\u001fA\t1b]2bY\u0006t\u0017\r^5wK*\t\u0011#A\u0002pe\u001e\u001c\u0001a\u0005\u0002\u0001)A\u0011Q\u0003G\u0007\u0002-)\tq#A\u0003tG\u0006d\u0017-\u0003\u0002\u001a-\t1\u0011I\\=SK\u001a\fa\u0001P5oSRtD#\u0001\u000f\u0011\u0005u\u0001Q\"\u0001\u0005\u0002\u00119,\u0007\u0010\u001e\'p]\u001e$\u0012\u0001\t\t\u0003+\u0005J!A\t\f\u0003\tUs\u0017\u000e\u001e\u0015\u0003\u0005\u0011\u0002\"!\n\u0015\u000e\u0003\u0019R!a\n\t\u0002\u000b),h.\u001b;\n\u0005%2#\u0001\u0002+fgR\fqA\\3yi&sG\u000f\u000b\u0002\u0004I\u0005Qa.\u001a=u\t>,(\r\\3)\u0005\u0011!\u0013a\u00038fqR\u0014un\u001c7fC:D#!\u0002\u0013\u0002\u000bM\u0004H.\u001b;)\u0005\u0019!\u0003"
    )
  ScalaInlineInfo: length = 0x22 (unknown attribute)
   01 00 00 06 00 10 00 0A 00 00 AB 00 0A 00 00 78
   00 0A 00 00 4A 00 0A 00 00 09 00 0A 00 00 BA 00
   0A 00
  ScalaSig: length = 0x3 (unknown attribute)
   05 02 00
```

</details>

3. If it's required that all the super interfaces are implemented in Scala Native, how come I did not have to add `RandomGenerator.SplittableGenerator` or `RandomGenerator.StreamableGenerator`? These are also implemented by `SpittableRandom`.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/SplittableRandom.html

Reply:
RandomGenerator was added in Java 17. Scala.js is testing it's code using only Java 8,11 and 16 
https://github.com/scala-js/scala-js/blob/225395005a56cf7c2a55a88f15ba689dd36aa76f/Jenkinsfile#L434-L437

There would be no direct reference, that's correct. However, the Scala compiler would know the whole hierarchy of types, including the implemented interfaces of a given symbol/type. I'm not sure why exactly sure at this point why only the RandomGenerator was needed. Maybe because methods we actually use are originally defined in RandomGenerator instead of SplittableRandom (interface default methods)

Reply:
> RandomGenerator was added in Java 17. Scala.js is testing it's code using only Java 8,11 and 16

Right, but even though they do not test it in CI, I can still use Scala.js `SpittableRandom` with JDK 17 without any problems.

Anyway, thank you, this helped me realize the problem. It's not the user/test JDK version that matters, but because we compile `SpittableRandom` itself on JDK 17 that causes the problem. Which is probably the interface default methods as you mentioned.